cclib/parser/turbomoleparser.py

```diff
@@ -149,9 +149,82 @@ def extract(self, inputfile, line):
                     self.set_attribute('atomcoords', atomcoords)
                     self.set_attribute('atomnos', atomnos)
 
+        # Frequency values in aoforce.out
+        #        mode               7        8        9       10       11       12
+        #
+        #      frequency          53.33    88.32   146.85   171.70   251.75   289.44
+        #
+        #      symmetry            a        a        a        a        a        a
+        #
+        #         IR               YES      YES      YES      YES      YES      YES
+        # |dDIP/dQ|   (a.u.)     0.0002   0.0000   0.0005   0.0004   0.0000   0.0000
+        # intensity (km/mol)       0.05     0.00     0.39     0.28     0.00     0.00
+        # intensity (  %   )       0.05     0.00     0.40     0.28     0.00     0.00
+        #
+        #        RAMAN             YES      YES      YES      YES      YES      YES
+        #
+        #   1   c           x   0.00000  0.00001  0.00000 -0.01968 -0.04257  0.00001
+        #                   y  -0.08246 -0.08792  0.02675 -0.00010  0.00000  0.17930
+        #                   z   0.00001  0.00003  0.00004 -0.10350  0.11992 -0.00003
+        if 'NORMAL MODES and VIBRATIONAL FREQUENCIES (cm**(-1))' in line:
+            vibfreqs = []
+            vibsyms = []
+            vibirs = []
+            vibdisps = []
+            while '****  force : all done  ****' not in line:
+                if line.strip().startswith('frequency'):
+                    freqs = [float(i.replace('i','-')) for i in line.split()[1:]]
+                    vibfreqs.extend(freqs)
+                    self.skip_line(inputfile, ['b'])
+                    line = next(inputfile)
+                    if line.strip().startswith('symmetry'):
+                        syms = line.split()[1:]
+                        vibsyms.extend(syms)
+
+                    self.skip_lines(inputfile, ['b', 'IR', 'dQIP'])
+                    line = next(inputfile)
+                    if line.strip().startswith('intensity (km/mol)'):
+                        irs = [self.float(f) for f in line.split()[2:]]
+                        vibirs.extend(irs)
+
+                    self.skip_lines(inputfile, ['intensity', 'b', 'raman', 'b'])
+                    line = next(inputfile)
+                    x, y, z = [], [], []
+                    while line.split():
+                        x.append([float(i) for i in line.split()[3:]])
+                        line = next(inputfile)
+                        y.append([float(i) for i in line.split()[1:]])
+                        line = next(inputfile)
+                        z.append([float(i) for i in line.split()[1:]])
+                        line = next(inputfile)
+
+                    for j in range(len(x[0])):

```

Comment:
Is this equivalent to `vibdisps.append(zip(x, y, z))`?

Reply:
No, `x[0] = [-0.01201, 0.00079, 0.01032, 0.01229, -0.22313, -0.00142]` so `x[0]` is basically an array of the x values (like in this [line](https://github.com/cclib/cclib/blob/master/data/Turbomole/basicTurbomole5.9/dvb_ir/aoforce.out#L870)) and for an entire block of 20 atoms we have `x` as an array consisting of 20 arrays `x[0],x[1] ... x[19]` which have the 6 values for 6 modes that are printed in a line. `y and z` are similar in this aspect. So zip(x,y,z) would just make an array where the zipped elements would themselves be arrays with x,y,z coordinates for different modes.

Reply:
True, but you could use it in the inner loop?